### ISSUE TITLE
The `stl` setting from `'fillchars'` is being displayed in ts-context when using `'statuscolumn'`

### ISSUE THREAD

**Issue by b0o** [nvim-treesitter/nvim-treesitter-context#396]
### Description

When I `:set fillchars=stl:#`, I'm seeing those chars inside of ts-context: 
![2024-02-16_12-28-44_region](https://github.com/nvim-treesitter/nvim-treesitter-context/assets/21299126/ea9af29d-97bb-456e-9628-c3fd3afd0294)

This happens when I am using a custom `statuscolumn`.

### Neovim version

v0.10.0-dev-2371+gd60eeacae4

### Expected behavior

Fillchars shouldn't be visible in ts-context.

### Actual behavior

Fillchars are visible in ts-context.

### Minimal config

```Lua
local plugins = {
  ts = "https://github.com/nvim-treesitter/nvim-treesitter",
  ts_context = "https://github.com/nvim-treesitter/nvim-treesitter-context",
  -- ADD ADDITIONAL PLUGINS THAT ARE _NECESSARY_ TO REPRODUCE THE ISSUE
}

for name, url in pairs(plugins) do
  local install_path = "/tmp/nvim/site/" .. name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system({ "git", "clone", "--depth=1", url, install_path })
  end
  vim.o.runtimepath = install_path .. "," .. vim.o.runtimepath
end

require("nvim-treesitter.configs").setup({
  ensure_installed = {
    "javascript",
  },
})
require("treesitter-context").setup({})

vim.o.number = true
vim.o.relativenumber = true
vim.o.statuscolumn = "%=%r%= "
vim.o.fillchars = "stl:#"
```

### Steps to reproduce

1. `nvim --clean -u minimal.lua`
2. open a file and check out the ts-context behavior

**Comment by lewis6991** [comment]
@luukvbaal

**Comment by luukvbaal** [comment]
https://github.com/neovim/neovim/pull/27501 will fix this for 0.10. I suppose we could fix it here as well by supplying `fillchar` to `nvim_eval_statusline()`.

**Comment by lewis6991** [comment]
Awesome, thanks for this.

I'll accept a PR to fix this here, otherwise I'll close this as an upstream issue.
